src/cli/plugin.rs

```diff
@@ -0,0 +1,116 @@
+use crate::serve_session::ServeSession;
+
+use memofs::InMemoryFs;
+use memofs::{Vfs, VfsSnapshot};
+use roblox_install::RobloxStudio;
+use snafu::{ResultExt, Snafu};
+use std::{
+    fs::{self, File},
+    io::{self, BufWriter},
+};
+use structopt::StructOpt;
+
+static PLUGIN_FILE_NAME: &str = concat!("rojo-", env!("CARGO_PKG_VERSION"), ".rbxmx");
+
+#[derive(Debug, StructOpt)]
+enum PluginSubcommand {
+    /// Install the plugin in Roblox plugins folder. If the plugin is already installed, installing
+    /// it again will overwrite the current plugin file.
+    Install,
+    /// Removes the plugin if installed.
+    Uninstall,
+}
+
+/// Install rojo's plugin.
+#[derive(Debug, StructOpt)]
+pub struct PluginCommand {
+    #[structopt(subcommand)]
+    subcommand: PluginSubcommand,
+}
+
+#[derive(Debug, Snafu)]
+pub struct PluginError(Error);
+
+#[derive(Debug, Snafu)]
+enum Error {
+    #[snafu(display("Could not locate Roblox Studio: {}", source.to_string()))]
+    CannotLocateRobloxStudio { source: roblox_install::Error },
+
+    #[snafu(display("{}", source))]
+    Io { source: io::Error },
+}
+
+fn xml_encode_config() -> rbx_xml::EncodeOptions {
+    rbx_xml::EncodeOptions::new().property_behavior(rbx_xml::EncodePropertyBehavior::WriteUnknown)
+}
+
+pub fn plugin(options: PluginCommand) -> Result<(), PluginError> {
+    match options.subcommand {
+        PluginSubcommand::Install => install_plugin()?,
+        PluginSubcommand::Uninstall => uninstall_plugin()?,
+    }
+    Ok(())
```

Comment:
Because both of these functions return `PluginError`, I think we can simplify this block to not need `?`:

```suggestion
    match options.subcommand {
        PluginSubcommand::Install => install_plugin(),
        PluginSubcommand::Uninstall => uninstall_plugin(),
    }
```